src/lib/containers/entity_finder/EntityFinder.tsx

```diff
@@ -1,6 +1,7 @@
 import { library } from '@fortawesome/fontawesome-svg-core'
 import { faSearch, faTimes } from '@fortawesome/free-solid-svg-icons'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { Map } from 'immutable'
```

Comment:
We use a [Map from Immutable.js](https://immutable-js.com/docs/v4.0.0-rc.15/Map/) to track selected entities because React won't update when an object is mutated, so using an immutable data structure is a bit more safe. The API of Immutable.js is also pretty powerful, letting us do things like batch mutations to avoid performance costs, e.g. copying the entire map when making a small change.